### DISCUSSION TITLE
[v3] updatedPage method doesn't work if we click on the back or forward button in the browser

### DISCUSSION THREAD

**Discussion by N1ebieski** [livewire/livewire#6305] [7. Bugs (archived)]
### Livewire version

3.0.0-beta.8

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/N1ebieski/laravel-livewire-v3.test/tree/pagination-updated-page-bug

### Steps To Reproduce

1. Clone my [repo](https://github.com/N1ebieski/laravel-livewire-v3.test/tree/pagination-updated-page-bug)
2. Switch to ```pagination-updated-page-bug``` branch
3. ```composer install && npm install && npm run build```
4. Go to /pagination-page
5. Click on the "next page button" in the pagination component
6. Livewire shows the dump message "Page has been updated!"
7. Click on the "back button" in the browser. Livewire will not show the dump message

https://github.com/livewire/livewire/assets/14810714/8a6c1b02-1b25-4c3c-be09-d43921baf1fe

### Additional information

Perhaps this is intentional but I think if a query param ```?page``` changes the property ```page``` in the component, the ```updatedPage``` method should have been called

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by C0chett0** [comment]
It actually makes sense.

When clicking on the pagination links button a call is made to the livewire component to update the value of the page hence the dump message appearing.

But when clicking back (or forward or juste plain typing the url with the page param) on the browser the component is initialized with the values from the query string directly, the page property is not updated it is only set once.

What can be confusing is seeing the different urls in your browser history since you didn't have a full round trip for any of them, it's just livewire that pushes the query to the memory of the browser. But keep in mind that using the navigation link an xhr call is made to /livewire/update which will trigger hooks but when clicking back/forward a full call is made to /pagination-page.

Hope it helps !

L.

PS : I stumbled on your issue because on my project pagination links works okay but never update the query string for some reason. I'll keep searching !

**Reply by N1ebieski** [reply]
Thanks for the answer.

> but never update the query string for some reason. I'll keep searching !

[Maybe you are using lazy component](https://github.com/livewire/livewire/discussions/5958)
